### PR TITLE
feat: added basic logger

### PR DESCRIPTION
**Pull request by FRALEXAlexander**

### CODE REVIEW

File: src/main/java/at/aau/utils/Logger.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package at.aau.utils;
+
+import java.util.ArrayList;
+
+public class Logger {
+    private static Logger logInstance;
+    private final ArrayList<String> logs = new ArrayList<>();
+
+
+    private Logger() {
```
Comment (TannerGabriel) [inline src/main/java/at/aau/utils/Logger.java:10]: Empty constructor is not needed

File: src/main/java/at/aau/utils/Logger.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package at.aau.utils;
+
+import java.util.ArrayList;
+
+public class Logger {
```
Comment (TannerGabriel) [inline src/main/java/at/aau/utils/Logger.java:5]: We should probably add some basic unit tests for the logger

File: src/main/java/at/aau/utils/Logger.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                logInstance = new Logger();
+            }
+        }
+        return logInstance;
+    }
+
+    public String getLogsString() {
+        String errorHeadline = "# ERRORS <br>\n";
+
+        synchronized (logs) {
```
Comment (TannerGabriel) [inline src/main/java/at/aau/utils/Logger.java:25]: We only execute `getLogsString` once all crawlers are finished. Do we really need the `synchronized`?

**Review by TannerGabriel — CHANGES_REQUESTED** [review]

File: src/main/java/at/aau/utils/Logger.java (hunk 3 above)
Comment (FRALEXAlexander) [reply src/main/java/at/aau/utils/Logger.java:25]: I think we could life without it but if the usage changes the syncronyced will help

**Review by TannerGabriel — APPROVED** [review]

**Pull request merged**